### PR TITLE
Thru line from Eligibility->Audit Info->Access steps to create an SF-SAC

### PR DESCRIPTION
**Pull request by Jkrzy**
We expand backend functionality to cover API request-response cycles across and stub out spaces for future enhancement of validation:

> Eligibility -> Auditee Info -> User Access Roles

A large change here is the addition of a `UserProfile` model, where we store user's responses to the initial forms leading up to the creation of a `SingleAuditChecklist`. 

This is an intentional decision to manage the state of the intake process on the backend, specifically for the Frontend being developed in `/FAC-Frontend`. Initially, we anticipate these end-points to be used only by that frontend to provide a multi-step user experience. 

Also included is a new model laying the foundations for #128. There's work to be done here to finalize the handling of inbound access, this PR gives us a starting point and is not intended as final decisions.

### CODE REVIEW

File: backend/audit/test_models.py [hunk 1]
```diff
@@ -9,6 +9,6 @@
 class SingleAuditChecklistTests(TestCase):
 
     def test_str_is_id_and_ein(self):
```
Comment (NicholasKramer) [inline backend/audit/test_models.py:11]: Minor: consider changing function name to "test_str_is_id_and_uei"

**Comment by NicholasKramer** [comment]
Awesome work so far @Jkrzy!

File: backend/audit/test_models.py (hunk 1 above)
Comment (Jkrzy) [reply backend/audit/test_models.py:11]: Thanks @NicholasKramer, let's use :octocat: suggestion for things like this.

```suggestion
    def test_str_is_id_and_uei(self):
```

**Review by NicholasKramer — APPROVED** [review]

**Comment by Jkrzy** [comment]
Thanks @NicholasKramer, I'll work on working resolving the conflicts here and ping you when ready for another quick look.

**Review by NicholasKramer — APPROVED** [review]
Nice work @Jkrzy!

**Pull request merged**